fluss-flink/fluss-flink-common/src/test/java/com/alibaba/fluss/flink/catalog/FlinkTableFactoryTest.java

```diff
@@ -0,0 +1,257 @@
+                .hasMessageContaining("Full lookup caching is not supported yet.");
+    }
+
+    @Test
+    void testSink() {
+        ResolvedSchema schema = createBasicSchema();
+        Map<String, String> properties = getBasicOptionsWithBucketKey();
+        properties.put(BUCKET_NUMBER.key(), "100");
+        FlinkTableSink tableSink = (FlinkTableSink) createTableSink(schema, properties);
+        List<String> bucketKeys = tableSink.getBucketKeys();
+        assertThat(bucketKeys)
+                .isEqualTo(Arrays.asList(properties.get(BUCKET_KEY.key()).split(",")));
+    }
+
+    private ResolvedSchema createBasicSchema() {
+        return new ResolvedSchema(
+                Arrays.asList(
+                        Column.physical("first", DataTypes.STRING().notNull()),
+                        Column.physical("second", DataTypes.INT()),
+                        Column.physical("third", DataTypes.STRING().notNull())),
+                Collections.emptyList(),
+                UniqueConstraint.primaryKey("PK_first_third", Arrays.asList("first", "third")));
+    }
+
+    private static Map<String, String> getBasicOptions() {
+        Map<String, String> options = new HashMap<>();
+        options.put("connector", "fluss");
+        options.put(BOOTSTRAP_SERVERS.key(), "0.0.0.1:9092");
+        return options;
+    }
+
+    private static Map<String, String> getBasicOptionsWithBucketKey() {
+        Map<String, String> basicOptions = getBasicOptions();
+        basicOptions.put(BUCKET_KEY.key(), "first");
+        return basicOptions;
+    }
+
+    private static DynamicTableSource createTableSource(
+            ResolvedSchema schema, Map<String, String> options) {
+        return createTableSource(schema, options, Collections.emptyMap());
+    }
+
+    private static DynamicTableSource createTableSource(
+            ResolvedSchema schema,
+            Map<String, String> options,
+            Map<String, String> enrichmentOptions) {
+        return FactoryUtil.createDynamicTableSource(
+                null,
+                OBJECT_IDENTIFIER,
+                new ResolvedCatalogTable(
+                        CatalogTable.of(
+                                Schema.newBuilder().fromResolvedSchema(schema).build(),
+                                "mock source",
+                                schema.getPrimaryKey()
+                                        .map(UniqueConstraint::getColumns)
+                                        .orElse(Collections.emptyList()),
+                                options),
+                        schema),
+                enrichmentOptions,
+                new Configuration(),
+                Thread.currentThread().getContextClassLoader(),
+                false);
+    }
+
+    private static DynamicTableSink createTableSink(
+            ResolvedSchema schema, Map<String, String> options) {
+        return FactoryUtil.createDynamicTableSink(
```

Comment:
In order to keep the tests in `FlinkCatalogITCase`, I think you can create TableSink and TableSource by 
1. first getting the Catalog via `FactoryUtil.createCatalog` 
2. and then get the `FlinkTableFactory` from `Catalog#getFactory()`.
3. last, create TableSource/TableSink via `createDynamicTableSource/Sink` method.